roles/run_hook/tasks/main.yml

```diff
@@ -29,32 +29,44 @@
         selectattr('key', 'match', _matcher)
       }}
     _single_hooks: >-
-      {% for _hook in _filtered_hooks -%}
-      - name: {{ _hook.key | regex_replace(_matcher, '\1') |
-                 replace('_', ' ') | capitalize }}
-      {{ _hook.value | to_nice_yaml | indent(width=2, first=true) }}
+      {% set hooks_list = [] -%}
+      {% for hook_element in (_filtered_hooks | default([])) -%}
+      {%   set hook_definitions = [hook_element] if hook_element is mapping else hook_element -%}
+      {%   for hook_definition in hook_definitions -%}
+      {%     set name = hook_definition.key | regex_replace(_matcher, '\\1') | replace('_', ' ') | capitalize -%}
+      {%     set _ = hooks_list.append(hook_definition.value | combine({'name': name})) -%}
+      {%   endfor -%}
       {% endfor -%}
+      {{ hooks_list }}
   block:
     - name: Assert parameters are valid
       ansible.builtin.assert:
         quiet: true
         that:
-          - _filtered_hooks is not string
-          - _filtered_hooks is not mapping
-          - _filtered_hooks is iterable
           - _list_hooks is not string
           - _list_hooks is not mapping
           - _list_hooks is iterable
           - (hooks | default([])) is not string
           - (hooks | default([])) is not mapping
           - (hooks | default([])) is iterable
 
+    - name: Assert single hooks are all mappings
+      vars:
+        _not_mapping_hooks: >-
+          {{
+            _single_hooks | reject('mapping') | list
+          }}
+      ansible.builtin.assert:
+        quiet: true
+        that:
+          - _not_mapping_hooks | length == 0
```

Comment:
maybe a clear message here? "At least one single hook doesn't seem to be a mapping" or something like that?